plugin/pkg/admission/podplacementrestriction/admission.go

```
@@ -0,0 +1,258 @@
+limitations under the License.
+*/
+
+package podplacementrestriction
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"reflect"
+
+	"k8s.io/kubernetes/pkg/admission"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/errors"
+	clientset "k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset"
+	"k8s.io/kubernetes/pkg/controller/framework"
+	"k8s.io/kubernetes/pkg/controller/framework/informers"
+	"k8s.io/kubernetes/pkg/util/yaml"
+)
+
+func init() {
+	admission.RegisterPlugin("PodPlacementRestriction", func(client clientset.Interface, config io.Reader) (admission.Interface, error) {
+		pluginConfig := readConfig(config)
+		return NewPodTolerationsPlugin(client, pluginConfig.PodTolerationsPluginConfig["clusterDefaultTolerations"]), nil
+	})
+}
+
+var _ = admission.Interface(&podTolerationsPlugin{})
+var _ = admission.WantsInformerFactory(&podTolerationsPlugin{})
+
+type podTolerationsPlugin struct {
+	*admission.Handler
+	client            clientset.Interface
+	namespaceInformer framework.SharedIndexInformer
+	// global default tolerations in a cluster. If a namespace is
+	// not assigned tolerations, it gets this by default.
+	clusterDefaultTolerations []api.Toleration
+}
+
+type pluginConfig struct {
+	PodTolerationsPluginConfig map[string][]api.Toleration
+}
+
+// readConfig reads default value of clusterDefaultTolerations
+// from the file provided with --admission-control-config-file
+// If the file is not supplied, it defaults to "".
+// The format in a file:
+// podTolerationsPluginConfig:
+//  clusterDefaultTolerations:
+//    - Key: key1
+//      Value: value1
+//    - Key: key2
+//      Value: value2
+func readConfig(config io.Reader) *pluginConfig {
+	defaultConfig := &pluginConfig{}
+	if config == nil || reflect.ValueOf(config).IsNil() {
+		return defaultConfig
+	}
+	d := yaml.NewYAMLOrJSONDecoder(config, 4096)
+	for {
+		if err := d.Decode(defaultConfig); err != nil {
+			if err != io.EOF {
+				continue
```

is there a reason you are ignoring the errors?